python/sqlflow_submitter/xgboost/feature_column.py

```diff
@@ -190,6 +226,46 @@ def categorical_column_with_hash_bucket(key, hash_bucket_size, dtype='string'):
                                                       dtype)
 
 
+class IndicatorColumnTransformer(BaseColumnTransformer):
+    def __init__(self, categorical_column):
+        assert isinstance(categorical_column, CategoricalColumnTransformer), \
+            "categorical_column must be type of CategoricalColumnTransformer but got {}".format(type(categorical_column))
+        self.categorical_column = categorical_column
+
+    def _set_field_names(self, field_names):
+        BaseColumnTransformer._set_field_names(self, field_names)
+        self.categorical_column._set_field_names(field_names)
+
+    def get_column_names(self):
+        return self.categorical_column.get_column_names()
+
+    def __call__(self, inputs):
+        slot = self.categorical_column(inputs)
+        assert len(
+            slot
+        ) == 1, "indicator_column does not accept sparse categorical feature"
+
+        def transform_fn(slot_value):
+            num_classes = self.categorical_column.num_classes()
+            if isinstance(slot_value, np.ndarray):
```

Comment:
Can we add comments that this `np.array` is only 2 dimensional?

Reply:
This `np.array` may be multi-dimensional. See [here](https://github.com/sql-machine-learning/sqlflow/pull/2416/files/456b7e3d91312b1ab0b8cb74bf97886ed2a3acad#diff-2f0634b1ebc2f4375900e5a40aaad875R255).

Reply:
Is slot_value also multi-dimensional?

Reply:
Yes!